packages/material-ui-styles/src/index.d.ts

```diff
@@ -305,15 +304,29 @@ declare module '@material-ui/styles/withStyles' {
 }
 
 declare module '@material-ui/styles/withTheme' {
-  import { PropInjector } from '@material-ui/core';
+  import * as React from 'react';
+  import { ConsistentWith, Omit, PropInjector } from '@material-ui/core';
+
+  export interface WithThemeCreatorOption<Theme> {
+    defaultTheme?: Theme;
+  }
+
+  export function withThemeCreator<Theme>(
+    option?: WithThemeCreatorOption<Theme>,
+  ): PropInjector<WithTheme<Theme>, Partial<WithTheme<Theme>>>;
 
   export interface WithTheme<Theme> {
     theme: Theme;
     innerRef?: React.Ref<any>;
   }
 
-  export default function withTheme<Theme>(): PropInjector<
-    WithTheme<Theme>,
-    Partial<WithTheme<Theme>>
+  export default function withTheme<
```

Comment:
I've struggled with this a couple of times but there's no way to define a generic const, right? For example I couldn't `declare const withTheme<Theme>: PropInjector<...>` (syntax is wrong but do you know the correct syntax for these semantics)? 

Only workaround I know so far is `export default interface WithTheme<T> extends PropInjector<...> {}` but while this may work it might be confusing to others. I'm ok with the duplication for now. Hard to gauge what costs more to maintain: duplication or breakage between typescript versions because that pattern only works incidental.